dali/test/python/test_operator_slice.py

```diff
@@ -701,3 +704,86 @@ def test_no_slice():
+        ]
+        if use_negative:
+            options += [
+                np.array([-2, -3], dtype=np.int32),
+                np.array([-2, 0], dtype=np.int32),
+                np.array([-3, 1], dtype=np.int32),
+                np.array([0, -2], dtype=np.int32),
+                np.array([1, -3], dtype=np.int32),
+                np.array([-2], dtype=np.int32),
+                np.array([-3], dtype=np.int32),
+            ]
+
+        axes = []
+        rel_start = []
+        rel_shape = []
+        for _ in range(batch_size):
+            axes_choice = random.choice(options)
+            axes += [axes_choice]
+            rel_start += [np.random.uniform(0.0, 0.2, axes_choice.shape)]
+            rel_shape += [np.random.uniform(0.4, 0.6, axes_choice.shape)]
+        return axes, rel_start, rel_shape
+
+    @pipeline_def(batch_size=batch_size, num_threads=num_threads, device_id=0)
+    def make_pipe():
+        encoded, _ = fn.readers.caffe(path=caffe_db_folder, random_shuffle=False)
+        image = fn.decoders.image(encoded, device="cpu", output_type=types.RGB)
+        if device == 'gpu':
+            image = image.gpu()
+        axes, rel_start, rel_shape = fn.external_source(source=get_dynamic_axes, num_outputs=3)
+        rel_start = fn.cast(rel_start, dtype=types.FLOAT)
+        rel_shape = fn.cast(rel_shape, dtype=types.FLOAT)
+        sliced1 = fn.slice(image, rel_start=rel_start, rel_shape=rel_shape, axes=axes)
+        sliced2 = fn.slice(image, rel_start, rel_shape, axes=axes)
+        return image, axes, rel_start, rel_shape, sliced1, sliced2
+    pipe = make_pipe()
+    pipe.build()
+    ndim = 3
+    for _ in range(3):
+        outs = pipe.run()
+        for sample_idx in range(batch_size):
+            in_img = np.array(
+                outs[0][sample_idx].as_cpu() if device == 'gpu' else
+                outs[0][sample_idx])
+            axes = np.array(outs[1][sample_idx])
+            rel_start = np.array(outs[2][sample_idx])
+            rel_shape = np.array(outs[3][sample_idx])
+            sliced1 = np.array(
+                outs[4][sample_idx].as_cpu() if device == 'gpu' else
+                outs[4][sample_idx]
+            )
+            sliced2 = np.array(
+                outs[5][sample_idx].as_cpu() if device == 'gpu' else
+                outs[5][sample_idx]
+            )
+            start = np.zeros([ndim], dtype=np.int32)
+            end = np.array([in_img.shape[i] for i in range(ndim)], dtype=np.int32)
+            for i in range(len(axes)):
+                a = axes[i]
+                assert a >= -ndim and a <= (ndim-1)
+                start[a] = roundint(rel_start[i] * in_img.shape[a])
+                end[a] = roundint((rel_start[i] + rel_shape[i]) * in_img.shape[a])
+            np.testing.assert_allclose(in_img[start[0]:end[0], start[1]:end[1], start[2]:end[2]], sliced1)
+            np.testing.assert_allclose(in_img[start[0]:end[0], start[1]:end[1], start[2]:end[2]], sliced2)
```

Comment:
Maybe it would save us some time if we not slice the `in_img` twice?
```suggestion
            np.testing.assert_allclose(sliced1, sliced2)
```